### ISSUE TITLE
Delta Force (2507950)

### ISSUE THREAD

**Issue by JoeyDayy** [ValveSoftware/Proton#7964]
# Compatibility Report
- Name of the game with compatibility issues: Delta Force: Hawk Ops
- Steam AppID of the game: 2709440

## System Information
- GPU:  NVIDIA RTX 3070
- Video driver version:  NVIDIA 555.58
- Kernel version:  Linux-zen-6.10.2
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/JoeyDayy/4e2e8feeb48f10cf12741cc8c83432d3/raw/7b0c14f1eaee3c8184f395fc2d2950ee0dfb3451/gistfile1.txt
https://gist.github.com/JoeyDayy/3c0b4f3f1e144c58048769d50288cc10#file-gistfile1-txt
- Proton version: 9.0-2

## I confirm:
- [ X ] that I haven't found an existing compatibility report for this game.
- [ X ] that I have checked whether there are updates for my system available.

https://gist.github.com/JoeyDayy/4bafff22bdb4af7e68c2982a79ff1fd8#file-gistfile1-txt

## Symptoms 
The launcher loads and is intractable however when selecting play the client either crashes or gets stuck on saying "running"in the launcher. However there is nothing happening. I did at one point get an error. 
![image](https://github.com/user-attachments/assets/6579c298-8e70-449e-b308-552396a29440)
However this does not appear most the time.

## Reproduction

Launch game with proton 9.0-2 or experimental. Click play on the bottom right of the launcher. Observe as it will either crash or say running without change.

**Comment by kisak-valve** [comment]
Delta Force Hawk Ops #2709440

Issue transferred from https://github.com/ValveSoftware/Proton/issues/7966.
@bitbucket90 posted on 2024-08-05T18:03:05:

# Compatibility Report
- Name of the game with compatibility issues: Delta Foce Hawk Ops
- Steam AppID of the game: 2709440

## System Information
- GPU: Nvidia 4080
- Video driver version: Driver Version: 555.58.02  
- Kernel version: 6.10.2-arch1-2
- Link to full system information report as [Gist](https://gist.github.com/):
- Proton version: PE, P8-P11 tests

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.


[steam-2709440.tar.gz](https://github.com/user-attachments/files/16501185/steam-2709440.tar.gz)




## Symptoms 
Launcher will launch and save state, but getting vulkan error and error log. 
![image](https://github.com/user-attachments/assets/c5829741-50c4-4641-97b0-c3b12c5266dd)



## Reproduction
In beta testing, just try to launch game to no avail. Launcher launches but will not go to game

**Comment by pollux78** [comment]
its weird a steamdeck user was able to get the game running while on my setup it just sticks to saying running but nothing ever appears, im on a rx 6700, cachyos so my chances of it running would be similar to a steamdeck but that doesnt seem the case

https://www.reddit.com/r/SteamDeck/comments/1em08yi/proof_of_delta_force_ho_on_steamos/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

**Comment by Sqwrrly** [comment]
> its weird a steamdeck user was able to get the game running while on my setup it just sticks to saying running but nothing ever appears, im on a rx 6700, cachyos so my chances of it running would be similar to a steamdeck but that doesnt seem the case
> 
> https://www.reddit.com/r/SteamDeck/comments/1em08yi/proof_of_delta_force_ho_on_steamos/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

I got it running on my Deck also. Proton Experimental. First time I tried Proton Experimental it opened the game but froze. I force closed it and tried again and it opened. Doesn't recognize the buttons/sticks so I had to use the touchpad in the menues. Closed it after that. On my desktop (Arch, Nvidia) I can't even get the launcher to open let alone the game. The launcher window just opens and closes.

**Comment by kodatarule** [comment]
The game works on Steam Deck completely fine, however it doesn't start at all on any linux distro, the logs show minimal differences.
With Proton GE 7.55 you can get quite close to the log on Steam Deck but it still fails to launch the game.
[steam-2709440cachyos.log](https://github.com/user-attachments/files/16560107/steam-2709440cachyos.log)
[steam-2709440SteamDeck.zip](https://github.com/user-attachments/files/16560110/steam-2709440SteamDeck.zip)

EDIT: I have tested multiple Proton versions which all result in the same outputs in the log and closing after launcher. Even tried to copy the steam deck prefix to no avail.

**Comment by McMarius11** [comment]
i think the anti cheat is blocking the launch of the game, hopefully they will support linux some day.

with the latest experimental 
![image](https://github.com/user-attachments/assets/38230cb0-1db7-4ea2-9251-4882d62ca45d)
when i click play the launcher will show running but the game will not start.
here is the steamlog
https://gist.github.com/McMarius11/e2669f68443b08094fb52193ec650e77

on windows after i click on play it wants to run
ACE Setup after that SGuardUpdate

Operating System: Manjaro Linux 
KDE Plasma Version: 6.0.5
KDE Frameworks Version: 6.4.0
Qt Version: 6.7.2
Kernel Version: 6.6.44-1-MANJARO (64-bit)
Graphics Platform: Wayland
Processors: 16 × AMD Ryzen 7 7800X3D 8-Core Processor
Memory: 62,4 GiB of RAM
Graphics Processor: AMD Radeon RX 7800 XT
Manufacturer: ASRock
Product Name: B650E PG-ITX WiFi

**Comment by DSDV** [comment]
Really wired "AntiCheat" which discriminates Desktop Penguins :( 
also in all other games form their brand seams to be a workaround with Hoyo/Collapse which is an alternative FOSS launcher. According to some statements (also readable on reddit) accepted by the Devs o.o

**Comment by SeeUnsharp** [comment]
Please vote in their official discord

https://discord.com/channels/1128946232207888414/1135844557133709465/1279063364743467169

**Comment by DSDV** [comment]
Invite-link: https://discord.gg/deltaforcegame

**Comment by jlrpalomar** [comment]
+1 here, unable to start it on Proton 8, 9 or experimental.  The launcher opens but after clicking launch it will simply get closed without any message.

Haven't tested in Steamdeck because I want to play this in PC. Voted on discord for it

It's Linux Mint 22 - Kernel 6.8.0

**Comment by robotman40** [comment]
Since it seems to run on Steam Deck, could anyone try using the Steam Deck environment variable? Maybe they added a special code path for Deck?

**Comment by DSDV** [comment]
~~i have no access yet but~~ i tried _dungeon born_ and the _splitgate 2 alpha_ which have the same "anti cheat" and the same problem (also only runs on the deck). 

i tired `SteamDeck=1 %command%` with no success :/

Also tested `SteamDeck=1  gamescope -W 1920 -H 1080 -r 60 -- %command%` **with the Delta Force Alpha** 

as many said only the launcher works the game crashes instant.

**Comment by ndavd** [comment]
The only protondb report that says they got it working on Deck mentions `Switch to older version: 7.0-6`

**Comment by DSDV** [comment]
you can see more examples from playable games: https://areweanticheatyet.com/?search=Anti-Cheat+Expert&sortOrder=&sortBy=

Dungeonborn for example (:

they all share "Anti-Cheat" "Expert" and all i checked run on the deck (fyi with terrible performance) bur not on Desktops.

All run fin on linux when you use hoyo other launchers based on hoyo (which as far as i understand simply don't load the expert anti cheat but are approved by the devs according to redit)

**Comment by blamedrop** [comment]
Wanted to create a new issue for `Delta Force (2507950)` but later I found out there is this one.
Game has separate app ID right now during alpha tests I guess.
This issue could be renamed IMO to:
> Delta Force (2507950) / Delta Force Alpha Test (2709440)

So it could be found from https://www.protondb.com/app/2507950 by 'GitHub Issue Search' button.

### System Information

Linux / Arch / kernel `6.10.7` / KDE / Wayland / `NVIDIA 560.35.03` / Proton Experimental [bleeding-edge]

### Symptoms

#### Launcher

 - mostly works, I could "log in" (it uses Steam account, but you have to select region/country), change some basic launcher settings
 - there is no music (not sure if there should be)
 - tray icon kinda works, although the right-click menu won't ever close (there is just one option there anyway - `Exit`)
 - sometimes the launcher will launch, hang, and crash
 - with `PROTON_LOG=1` it hangs and log gets quickly to a gigabyte, here's first 1024 lines of it: [1.log](https://github.com/user-attachments/files/16841914/1.log)
 - with `WINEDEBUG=-unwind` added (or without logging at all) launcher often freezes with this error, also see the log until this happened: [2.log](https://github.com/user-attachments/files/16841975/2.log)

![image](https://github.com/user-attachments/assets/d92e9a22-004a-447a-b88d-19e93c6f8bfe)

 - here's a log (with `WINEDEBUG=-unwind`), where the above error didn't happen and I clicked `LAUNCH` button - after clicking it the launcher disappeared, presumably game tried to launch, launcher reappeared and it just crashed, log: [3.log](https://github.com/user-attachments/files/16842005/3.log)

#### Actual game

Did not launch.
Sometimes I got launcher to say 'RUNNING' but no window appeared.

**Comment by Stang51** [comment]
Weapon / Ammo Management Help Please
Hi Guys,

I'm a old guy that has a super newbie question. When preparing my load out for a mission, I always get a message that says "Your Ammo is low". This is after I went to my stash, double clicked on 10 boxes of 5.56 ammo from my stash and 6 boxes of 9mm ammo, When I double click on the ammo, it adds them to my load out in the Slots section and sometimes to the Chest Rig section.. Sometimes it shows them only 1 box to a slot, sometimes it stack them and shows a number over that square like 10. Finally I drag a box on top of my rifle and it changes to say 1/17. When I get in game, I have 1 round in the rifle. When I hit R to reload, it says I have no ammo.

I know I'm doing something wrong.

Any help is appreciated.

Tejay

**Comment by LethalManBoob** [comment]
still an issue here
[steam-2951290.log](https://github.com/user-attachments/files/17364873/steam-2951290.log)

**Comment by DSDV** [comment]
can someone maybe test on HoloISO and/or ChimeraOS ? 

i got a friends PC specially for this event but i forgot it had an ** nvidia card :( 
so no gamescope and no test :( only black screens.

**Comment by NasirA1991** [comment]
If someone can figure out how to bypass the launcher then we might be able to actually launch the game .exe hmm

**Comment by pollux78** [comment]
> If someone can figure out how to bypass the launcher then we might be able to actually launch the game .exe hmm

when launching with deltaforceclient.exe it simply launches and stops after a couple of seconds

here is that log 
[steam-11817271922754846720.log](https://github.com/user-attachments/files/17366997/steam-11817271922754846720.log)

then here is the launcher log
[steam-2951290.log](https://github.com/user-attachments/files/17367006/steam-2951290.log)

**Comment by ndavd** [comment]
They are supposedly "cross platform", I don't understand how they are not supporting Linux

**Comment by pollux78** [comment]
> They are supposedly "cross platform", I don't understand how they are not supporting Linux

If that was the case then they should be supporting Mac os but you don't see that happening either 

Cross platform just means console and PC(windows)

They should be supporting the switch also but we don't see that either most likely because of its rather low end hardware 

I'm wondering if this is an anticheat problem or simply a wine/proton issue when trying to launch the game

**Comment by DSDV** [comment]
apparently they "consider" linux: https://www.reddit.com/r/pcgaming/comments/1er6s48/comment/li1m5yq/
i guess it is really on thier list since if you combine all the feature request / suggestion upvotes in thier discord linux is among the top. 

but i wont get my hopes up. i would rather bet on the alternative hoyo launchers ^^ (which cant use anti cheat - no argument what so ever xD)

**Comment by blamedrop** [comment]
For reference, these apps apply here:

- `Delta Force` (`2507950`)
- `Delta Force Alpha Test` (`2709440`)
- `Delta Force Demo` (`2951290`)

**Comment by AQDuck97** [comment]
Some people are claiming to run it on this Next Fest demo but I haven't seen any evidence of it

**Comment by data-tim** [comment]
Still no luck running with any version of Proton (Experimental or GE builds).

Can get the launcher but immediately closes when attempting game launch. (Fedora 40 - Gnome - Wayland - Nvidia)

**Comment by DSDV** [comment]
Anyone able to test with holoiso and or chimera os? 
sadly i could not get them to work on a 1050ti and on my mainpc i could not get them to boot from usb :(

**Comment by ndavd** [comment]
We aren't even sure if this is an issue with their anti-cheat or something else?

**Comment by DSDV** [comment]
i dont know but according to this issue and proton db / reddit there are folks who say it runs on the deck but not any other linux distro. 
So on nobara discord it was suggested to test with holoiso &  chimeraos to see if it runs there. And maybe see if we can "fix" other distros. 
but i dont own a deck and as written above i cant get both distros to run on my test / main hardware to do the testing myself :(

**Comment by ndavd** [comment]
Indeed, there's a recent report saying that it works on Deck https://www.protondb.com/app/2507950?device=steamDeck

**Comment by ErikReider** [comment]
Tested it right now on my deck (LCD) and the game boots. Doesn't seem to have controller support though so haven't played a actual match except for tactical turmoil which seems to work at a decent FPS :)

**Comment by ndavd** [comment]
I have no experience debugging these kinds of issues but perhaps a good approach would be to compare the Proton logs of someone who successfully launched the game on Deck with someone who failed to launch the game on Linux desktop to see what differs and fails

**Comment by LethalManBoob** [comment]
> I have no experience debugging these kinds of issues but perhaps a good approach would be to compare the Proton logs of someone who successfully launched the game on Deck with someone who failed to launch the game on Linux desktop to see what differs and fails

I have been told its to do with the non oled steam deck being manually approved for use in the game. so i assume it could possibly be a means of just spoofing the detection

**Comment by ndavd** [comment]
That sounds like they're intentionally whitelisting devices? So they're not supporting Linux intentionally?

**Comment by pollux78** [comment]
> That sounds like they're intentionally whitelisting devices? So they're not supporting Linux intentionally?

Yep seems like it, I was thinking of some Dev could figure out spoofing that we are all steamdeck LCDs 😂 

Not cool tho that they are manually doing this, very interesting and wonder what the reason is as steamdeck has a desktop mode so it's not very different from Linux desktop and it uses the same proton as everyone else

**Comment by ndavd** [comment]
I have been trying several things to emulate the Steam Deck when running the game:

- Running it through `gamescope`:
```
gamescope -w 1280 -h 800 -f -- %command%
```

- Running the Deck version of Steam:
  - Change `.steam/package/beta`'s contents to `steampal_stable_9a24a2bf68596b860cb6710d9ea307a76c29a04d`
  - Open Steam with the command `steam -gamepadui`
  
The only noticeable change in both of these methods is that instead of crashing the game it says `RUNNING...`, but nothing actually opens.

**Comment by ndavd** [comment]
In any case the demo ends tomorrow... Maybe something will change once the game actually releases in early access.

**Comment by DeathTBO** [comment]
Hello, I tried with a few different versions of Proton. Experiment, 9, GE 9, and 7.06. I get login failed a few times, but after the second try it lets me in. The game never seems to launch though. The log file PROTON_LOG=1 is massive 2.1gb. I truncated it down because it just repeats the same call stack over and over.

Fedora 42, 5900x, Radeon 5700xt, Mesa 24.2.8 - Kernel 6.11.10

[steam-2507950.log](https://github.com/user-attachments/files/18016833/steam-2507950.log)

**Comment by lzq960605** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/7964#issuecomment-2519010246

Only steam deck

**Comment by lzq960605** [comment]
Steam deck oled can run without any problems

**Comment by pollux78** [comment]
> Steam deck oled can run without any problems

I don't like when studios exclude Linux desktop when steamdeck is literally a x86 PC and can do the same thing as a Linux desktop PC 

The mods on the discord say linux is not supported not even on the steamdeck but the game works without issues on steamdeck, quite odd if they didnt actually intend on supporting the steamdeck 

If so wonder what is causing this game not to launch properly and it would be awesome if valves engineers could chime in here if they know why linux desktop wont work with this game but steamdeck will

**Comment by DSDV** [comment]
sadly i have no system where i can test chimeraos or holoiso 
if someone can get this tested on those os would be really nice since they try to "simulate" steamos in a way (:

**Comment by kleinermob** [comment]
Is there a way to spoof it running on Steam deck?

**Comment by DSDV** [comment]
https://github.com/ValveSoftware/Proton/issues/7964#issuecomment-2323181169
afaik only the SteamDeck=1 option. 
I read somewhere about a protonhide but i dont see a sense it that. 
so far nobody figured out the diffrence. 

i tried to get HoloISO and ChimeraOS running last beta but sadly where unsuccessful in getting them to run on my testrig. 
If someone can test those OS on a spare Drive or Partition it would be interesting to see if the game works there since they try to "mimic" SteamOS

**Comment by rejedai** [comment]
Tried `SteamDeck=1 %command%` crash after pressing launch

**Comment by ruifcastro** [comment]
Just to be clear, it only runs on LCD Steam Decks (I have played about 2 hours on my Steam Deck). Does not run on the OLED Steam Decks (brother and friend can't play this game on the Steam Deck). It's the same thing that happens with Metal Slug Awakening (I have over 200 hours on the Steam Deck) which is from the same publisher TiMi Studio Group. 

I have Steam OS 3.6.20. I have the 256mb version. My brother and friend have the 512mb OLED version

**Comment by lzq960605** [comment]
> Just to be clear, it only runs on LCD Steam Decks (I have played about 2 hours on my Steam Deck). Does not run on the OLED Steam Decks (brother and friend can't play this game on the Steam Deck). It's the same thing that happens with Metal Slug Awakening (I have over 200 hours on the Steam Deck) which is from the same publisher TiMi Studio Group.
> 
> I have Steam OS 3.6.20. I have the 256mb version. My brother and friend have the 512mb OLED version

The oled version can run, but it was not possible before

**Comment by ndavd** [comment]
Any updates on this?

**Comment by TotalTaxAmount** [comment]
Setting LD_PRELOAD="" stops the game from crashing but it seems to just hang forever.

I created another suggestion to allow support for non steamos devices: https://discord.com/channels/1128946232207888414/1135844557133709465/1363762033258139771

**Comment by ruifcastro** [comment]
My brother and friend on the OLED version couldn't get it to run but after many updates the game now runs on the Steam Deck OLED models

**Comment by sklynic** [comment]
One of this game's developer (game director) have explicitly denied linux (proton) support. (game's behavior of running on native steam deck hardware is unchanged for now, though.)

source: `https://www.reddit.com/r/gaming/comments/1k8evyo/comment/mpc3vcd/`

edit: found this message here in this issue. let me post it again, I have no comment on this. `https://www.reddit.com/r/pcgaming/comments/1er6s48/comment/li1m5yq/`

**Comment by DSDV** [comment]
Would be interesting to see of the Lenovo device with official SteamOS also works 
or if it is actually somehow bound to hardware.

**Comment by ndavd** [comment]
As the user `tripod` reported on [protondb](https://www.protondb.com/app/2507950) days ago, the game does launch since `GE-Proton10-5`:
> Tinker Steps: Custom Proton: GE-Proton10-5, Set launch options
> gamemoderun %command
>
> Okay, I'm kind of on the fence about this, but if you use GE-Proton10-5 you can succesfully play the game no dramas. Obviously there is risk of account getting nuked due to ACE has only steam decks whitelisted

Unfortunately, at least for me (tried `GE-Proton10-6` and `GE-Proton10-8`), the anti-cheat crashes the game right during the splash screen:
<img width="324" height="145" alt="Image" src="https://github.com/user-attachments/assets/78d5224c-bad6-4b7e-a7e3-82c9ceef7dd0" />

**Comment by ndavd** [comment]
There are more reports that it is working in latest Proton versions, but I still couldn't get past the splash screen.

**Comment by sklynic** [comment]
Proton EM contains patches that bypass this game's anti-cheat client which obviously violates this game's ToS and can leads to ban, please refrain yourself from trying it out.

**Comment by ndavd** [comment]
> Proton EM contains patches that bypass this game's anti-cheat client which obviously violates this game's ToS and can leads to ban, please refrain yourself from trying it out.

Personally I don't really care myself if I get banned, since I could never play anyway so far. Does Proton EM stand for Experimental? I've tried even the Bleeding Edge but got no success. 

UPDATE: I found it, couldn't get to try out `EM Proton 10.0.23`: it says in the latest release that `Proton-EM 10.0-23 has been deleted because it contains patches that are no longer public`, which I guess might refer to the patches related to Delta Force, since in `EM Proton 10.0.24` I couldn't launch it, it says that `The driver could not be loaded`.

**Comment by pollux78** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/7964#issuecomment-3058307707

Pretty sure EM stands for the developers name Etaash-mathamsetty

**Comment by ndavd** [comment]
Yes, that's what I've tried.

**Comment by ndavd** [comment]
Can confirm the patches have been removed in `EM Proton 10.0.24` https://github.com/Etaash-mathamsetty/Proton/issues/18#issuecomment-3062618858

**Comment by BananaWorks07** [comment]
> Proton EM contains patches that bypass this game's anti-cheat client which obviously violates this game's ToS and can leads to ban, please refrain yourself from trying it out.

It's not a bypass btw, but either way you should not use it

**Comment by kleinermob** [comment]
> 
> UPDATE: I found it, couldn't get to try out `EM Proton 10.0.23`: it says in the latest release that `Proton-EM 10.0-23 has been deleted because it contains patches that are no longer public`, which I guess might refer to the patches related to Delta Force, since in `EM Proton 10.0.24` I couldn't launch it, it says that `The driver could not be loaded`.

seems like EM Proton 10.0.23 didnt have any special patch for Delta Force because im using it right now and also says "The driver could not be loaded"
or it was patched

**Comment by ndavd** [comment]
> seems like EM Proton 10.0.23 didnt have any special patch for Delta Force because im using it right now and also says "The driver could not be loaded"

That's odd, because there are multiple reviews on protondb that say it's working for them. But either way, I still haven't gotten any replies as to why the patches have been removed or what were they doing exactly in order to make it work. Probably was patched, yeah.

**Comment by BananaWorks07** [comment]
> > seems like EM Proton 10.0.23 didnt have any special patch for Delta Force because im using it right now and also says "The driver could not be loaded"
> 
> That's odd, because there are multiple reviews on protondb that say it's working for them. But either way, I still haven't gotten any replies as to why the patches have been removed or what were they doing exactly in order to make it work. Probably was patched, yeah.

Yes the game got updated, newer version of ACE no longer works with the original patches. Tbh Delta Force seems like a fun game, might download it and play with the updated patches ;)

**Comment by ndavd** [comment]
> Yes the game got updated, newer version of ACE no longer works with the original patches. Tbh Delta Force seems like a fun game, might download it and play with the updated patches ;)

Oh, I see. It does seem quite fun, shame they aren't interested in supporting Linux desktop.

**Comment by NjlsShade** [comment]
If the initial game launcher fails to start with a "4/77" error or any other network related messages, set the "PROTON_SET_GAME_DRIVE" variable to true.

Just putting this here as it looks to be a problem for some users.

**Comment by lxwulf** [comment]
Could it be that the standalone installer does not work at all on proton? The installers pop up but then does not download the game/launcher and hangs in the “initializing” status with 0%.

I already set up the environment variable with `PROTON_SET_GAME_DRIVE=1` but did not help…

I tested the latest proton version (bleeding edge) and `ge-proton10-25`.
